core/src/test/java/org/apache/iceberg/TestTables.java

```diff
@@ -103,6 +104,11 @@ public static TestTable load(File temp, String name) {
     return new TestTable(ops, name);
   }
 
+  public static TestTable load(File temp, String name, BiFunction<File, String, TestTableOperations> opsSupplier) {
```

Comment:
Since we only have one usage of this function, maybe instead of providing the fully extensible version here we provide a method like

```java
public static TestTable tableWithCommitStateUnknown(File temp, String name) {
commitStateUnknownOperations = new TestTables.TestTableOperations(name, file) {
          @Override
          public void commit(TableMetadata base, TableMetadata updatedMetadata) {
            super.commit(base, updatedMetadata);
            throw new CommitStateUnknownException(new RuntimeException(errorMsg));
          }
return new testTable(commitStateUnknownOperations, name)
}
```

Just since we aren't really using the BiFunction for anything but injecting this particular operations